src/ReactionNetwork.jl

```diff
@@ -0,0 +1,341 @@
+    end
+    return (Tuple(rates),Tuple(affects))
+end
+
+#From the tuples created in get_jump_expr, generates an expression which when evaluated will become a tuple of ConstantRateJumps to be used for Guillespie Simulations.
+function get_jumps(rates::Tuple, affects::Tuple,reactants::OrderedDict{Symbol,Int64},parameters::OrderedDict{Symbol,Int64})
+    jumps = Expr(:tuple)
+    for i = 1:length(rates)
+        push!(jumps.args,Expr(:call,:ConstantRateJump))
+    end
+    for i = 1:length(rates)
+        push!(jumps.args[i].args, :((u,p,t) -> $(recursive_replace!(deepcopy(rates[i]), (reactants,:u), (parameters, :p)))))
+        push!(jumps.args[i].args, :(integrator -> $(expr_arr_to_block(deepcopy(affects[i])))))
+    end
+    return jumps
+end
+
+#Recursively traverses an expression and removes things like X^1, 1*X. Will not actually have any affect on the expression when used as a function, but will make it much easier to look at it for debugging, as well as if it is transformed to LaTeX code.
+function recursive_clean!(expr::Any)
+    (typeof(expr)!=Expr) && (return expr)
+    for i = 1:length(expr.args)
+        expr.args[i] = recursive_clean!(expr.args[i])
+    end
+    (expr.args[1] == :^) && (expr.args[3] == 1) && (return expr.args[2])
+    if expr.args[1] == :*
+        for i = length(expr.args):-1:2
+            (expr.args[i] == 1) && deleteat!(expr.args,i)                   #Removes all multiplications by 1.
+        end
+        (length(expr.args) == 2) && (return expr.args[2])                   # We have a multiplication of only one thing, return only that thing.
+        (length(expr.args) == 1) && (return 1)                              #We have only * and no real argumenys.
+    end
+    if expr.head == :call
+        in(expr.args[1],hill_name) && return hill(expr)
+        (in(expr.args[1],mm_name)) && (return mm(expr))
+    end
+    return expr
+end
+
+#Recursively traverses an expression and replace instances of variables and parmaters with things that the DifferentialEquations packakes simulation algorithms can understand. E.g. X --> u[1], kB1 --> p[1] etc.
+function recursive_replace!(expr::Any, replace_requests::Tuple{OrderedDict{Symbol,Int64},Symbol}...)
+    if typeof(expr) == Symbol
+        for rr in replace_requests
+            (haskey(rr[1],expr)) && (return :($(rr[2])[$(rr[1][expr])]))
+        end
+    elseif typeof(expr) == Expr
+        for i = 1:length(expr.args)
+            expr.args[i] = recursive_replace!(expr.args[i], replace_requests...)
+        end
+    end
+    return expr
+end
+
+#Makes the Jacobian.
+function calculate_jac(f_expr::Vector{Expr}, syms)
+    symjac = Matrix{SymEngine.Basic}( length(syms), length(syms))
+    symfuncs = [SymEngine.Basic(f) for f in f_expr]
+    for i in eachindex(f_expr)
+        for j in eachindex(syms)
+          symjac[i,j] = diff(symfuncs[i],syms[j])
```

Comment:
just for reference: this looping is done in row major form so you usually want to flip the ordering in Julia. Performance won't matter here, but it's something to keep in mind.

Reply:
I just noticed that mistake is from my code 📦

Reply:
yup, we stole that and the type definition from ParameterizedFunctions.jl